### PR TITLE
Bugfix for double ";" when using function "putvalue"

### PR DESCRIPTION
**Pull request by louiswouters**
Fixed double ";" when used on an existing line with a comment.
Optimized the construction of the new G-code line by using list.append and join
Updated description (outdated)

### CODE REVIEW

File: plugins/PostProcessingPlugin/Script.py [hunk 1]
```diff
@@ -141,52 +141,46 @@ def putValue(self, line: str = "", **kwargs) -> str:
...
-            result += key + str(value)
-
-        #Put the comment back in.
-        if comment != "":
-            if result != "":
-                result += " "
-            result += ";" + comment
-
-        return result
+        def putValue(self, line: str = "", **kwargs) -> str:
```
Comment (konskarm) [inline plugins/PostProcessingPlugin/Script.py:152]: putValue(..) is already defined in line 136. Please remove line 152 and unindent the rest of the function.

**Comment by Ghostkeeper** [comment]
We already have a post-processing script to display progress in the LCD. If there's something you want to improve on that, please improve the existing one. It's also good to build separate features in separate pull requests, lest they block each other.

**Comment by louiswouters** [comment]
O, it appended this pull request... I tried to create a separate pull request but it looked like nothing had happened.

**Comment by konskarm** [comment]
@louiswouters, as @Ghostkeeper mentioned, it seems you are doing two thing in this PR, fixing the bug mentioned in the title and also adding a ShowProgress post-processing script (which we already have as DisplayProgressOnLCD). Please split these two into separate PR's and if needed modify the DisplayProgressOnLCD instead of adding an extra script.

**Pull request closed without merging**